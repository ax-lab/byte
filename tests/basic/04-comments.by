#! /usr/bin/byte

# This is a single line comment

#------------------------------------------------------------------------------#
# Boxes are pretty
#------------------------------------------------------------------------------#

# print 'this commented code will be ignored'

print 'first line'  # <-- first line that is actually printed


#(
	This is a multiline comment (parenthesis in(side) here are balanced)
)

#((
	Since parenthesis are balanced, you can start with any number!
))

#(((
	This style is more balanced, if it strikes your fancy.	
#)))

#(
	As ugly as this is, it will actually be printed:
)print 'second line'

# (
#   This style of comment allows easily toggling sections of code by
#   adding or removing a space before the parenthesis!

print 'third line'    # this is printed as long as the space is there

#   This would be the end of the disabled section!
# )

print '# fourth line is not a comment'

#------------------------------------------------------------------------------#
# The following tests are for indentation corner cases
#------------------------------------------------------------------------------#

#(
	The lexer should not consider this an indentation:

	)print 'fifth line is not indented'

if 1 == 1:
	#(

Indentation inside comments is ignored!

This inherits the indentation of the comment line:
	
	)print 'but sixth line is'

if 1 == 1:
	#(
		This is also indented because the original comment is:

)print 'seventh line'
