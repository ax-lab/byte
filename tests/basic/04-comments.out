first line
second line
third line
# fourth line is not a comment
fifth line is not indented
but sixth line is
seventh line
